lib/MusicBrainz/Server/Data/Work.pm

```diff
@@ -425,6 +426,66 @@ sub _find_recording_artists
     }
 }
 
+=method load_median_length
+
+This method will load the median length for the recordings of the work,
+ignoring any recordings marked as partial or as being a medley.
+
+For context, see the only recording-work relationship type, "performance",
+and its attributes:
+https://musicbrainz.org/relationship/a3005666-a872-32c3-ad06-98af558e99b0
+
+=cut
+
+sub load_median_length
+{
+    my ($self, @works) = @_;
+
+    my @ids = map { $_->id } @works;
+    return () unless @ids;
+
+    my %map = object_to_ids(@works);
+    $self->_find_median_length(\@ids, \%map);
+}
+
+sub _find_median_length
+{
+    my ($self, $ids, $map) = @_;
+    return unless @$ids;
+
+    my $query = "
+        SELECT lrw.entity1 AS work_id, median(r.length) AS length
+        FROM l_recording_work lrw
+        JOIN recording r ON lrw.entity0 = r.id
+        WHERE lrw.entity1 IN (" . placeholders(@$ids) . ")
+        AND NOT EXISTS (
+            SELECT 1
+            FROM l_recording_work lrw2
+            WHERE lrw2.entity0 = lrw.entity0
+            GROUP BY lrw2.entity0
+            HAVING COUNT(*) > 1
+        )
+        AND NOT EXISTS (
+            SELECT 1
+            FROM link_attribute la
+            WHERE la.link = lrw.link
+            AND la.attribute_type IN (
+                567, -- cover
+                579, -- partial
+                750 -- medley
```

Comment:
Didn’t we have some sort of standard comment to track `id` usage instead of `gid` in code?

Reply:
Did we? I'm honestly not sure, but happy to change if you or @mwiencek remember.